### PR TITLE
Added More Information About Adadelta Optimizer

### PR DESCRIPTION
**Pull request by hisushanta**
I have added more information about Adadelta Optimizer to developers understand faster ways to what is doing that.
It's my changes code looks like this:
Screenshot from 2023-07-31 10-01-54

### CODE REVIEW

File: torch/nn/modules/loss.py [hunk 1]
```diff
@@ -1763,6 +1764,63 @@ def forward(self, log_probs: Tensor, targets: Tensor, input_lengths: Tensor, tar
         return F.ctc_loss(log_probs, targets, input_lengths, target_lengths, self.blank, self.reduction,
                           self.zero_infinity)
 
+
+# This Loss function can be used for Object detection tasks.
+class ObjectDetectionCoordinatesAndSizeLoss(_Loss):
```
Comment (vadimkantorov) [inline torch/nn/modules/loss.py:1769]: is addition of this by mistake?

Comment (hi-sushanta) [reply torch/nn/modules/loss.py]: I apologize for the mistake. I have also removed the code from that branch.

File: torch/optim/adadelta.py [hunk 2]
```diff
@@ -162,6 +169,17 @@ def step(self, closure=None):
         {foreach}
         {maximize}
         {differentiable}
+        
+    The `rho` parameter controls the rate at which the squared gradients are
+    averaged. A higher value of `rho` will result in a slower average, which
```
Comment (janeyx99) [inline torch/optim/adadelta.py:174]: Can this description of rho be moved to where rho is introduced as an arg?

File: torch/nn/modules/__init__.py [hunk 3]
```diff
@@ -44,7 +44,7 @@
     'Tanhshrink', 'RReLU', 'L1Loss', 'NLLLoss', 'KLDivLoss', 'MSELoss', 'BCELoss', 'BCEWithLogitsLoss',
     'NLLLoss2d', 'PoissonNLLLoss', 'CosineEmbeddingLoss', 'CTCLoss', 'HingeEmbeddingLoss', 'MarginRankingLoss',
     'MultiLabelMarginLoss', 'MultiLabelSoftMarginLoss', 'MultiMarginLoss', 'SmoothL1Loss', 'GaussianNLLLoss',
-    'HuberLoss', 'SoftMarginLoss', 'CrossEntropyLoss', 'Container', 'Sequential', 'ModuleList', 'ModuleDict',
+    'HuberLoss', 'SoftMarginLoss', 'CrossEntropyLoss','Container', 'Sequential', 'ModuleList', 'ModuleDict',
```
Comment (janeyx99) [inline torch/nn/modules/__init__.py:47]: This change also seems like an accident

**Review by janeyx99 — COMMENTED** [review]
Thanks for taking the time to improve our docs! I've left some comments to make the docs more concise and relevant.

File: torch/nn/modules/__init__.py (hunk 3 above)
Comment (hi-sushanta) [reply torch/nn/modules/__init__.py:47]: But I don't see that mistake in my file.

**Comment by hi-sushanta** [comment]
If anyone can review my work?

File: torch/optim/adadelta.py [hunk 4]
```diff
@@ -149,20 +149,36 @@ def step(self, closure=None):
 
     For further details regarding the algorithm we refer to `ADADELTA: An Adaptive Learning Rate Method`_.
     """ + r"""
+    The `Adadelta` optimizer is a popular choice for deep learning, as it is
+    relatively simple to implement and can be effective in a variety of
+    settings.`Adadelta` is based on the idea of adaptive learning rates,
+    which means that the learning rate for each parameter is adjusted
+    individually based on its own history of gradients. This strategy can
+    increase robustness against changes in the data and prevent the optimizer
+    from getting stuck in local minima.
```
Comment (janeyx99) [inline torch/optim/adadelta.py:158]:
```suggestion
    The `Adadelta` optimizer is based on the idea of adaptive learning rates,
    which means that the learning rate for each parameter is adjusted
    individually based on its own history of gradients. This strategy can
    increase robustness against changes in the data and prevent the optimizer
    from getting stuck in local minima.
```

Comment (janeyx99) [reply torch/optim/adadelta.py:158]: Let's not make subjective claims in our docs about any particular optimizer

File: torch/optim/adadelta.py [hunk 5]
```diff
@@ -149,20 +149,36 @@ def step(self, closure=None):
...
         eps (float, optional): term added to the denominator to improve
-            numerical stability (default: 1e-6)
+            numerical stability (default: 1e-6). The `eps` parameter is a small value that is added to the denominator to
+            improve numerical stability.
         lr (float, optional): coefficient that scale delta before it is applied
-            to the parameters (default: 1.0)
-        weight_decay (float, optional): weight decay (L2 penalty) (default: 0)
+            to the parameters (default: 1.0). The `lr` parameter controls the overall
+            learning rate of the optimizer. A higher value of `lr` will result in
+            faster learning, but it can also lead to overfitting.
```
Comment (janeyx99) [inline torch/optim/adadelta.py:173]:
```suggestion
            learning rate of the optimizer.
```
It could also lead to nonconvergence. Since the lr is general across all optimizers, let's not make the claim in this specific optimizer.

File: torch/optim/adadelta.py [hunk 6]
```diff
@@ -149,20 +149,36 @@ def step(self, closure=None):
...
         rho (float, optional): coefficient used for computing a running average
-            of squared gradients (default: 0.9)
+            of squared gradients (default: 0.9). The `rho` parameter controls the 
+            rate at which the squared gradients are averaged. A higher value of 
+            `rho` will result in a slower average, which can be helpful for preventing 
+            oscillations in the learning process.
         eps (float, optional): term added to the denominator to improve
-            numerical stability (default: 1e-6)
+            numerical stability (default: 1e-6). The `eps` parameter is a small value that is added to the denominator to
+            improve numerical stability.
```
Comment (janeyx99) [inline torch/optim/adadelta.py:169]: Redundant
```suggestion
            numerical stability (default: 1e-6).
```

File: torch/optim/adadelta.py [hunk 7]
```diff
@@ -149,20 +149,36 @@ def step(self, closure=None):
...
+    from getting stuck in local minima.
     Args:
         params (iterable): iterable of parameters to optimize or dicts defining
             parameter groups
         rho (float, optional): coefficient used for computing a running average
-            of squared gradients (default: 0.9)
+            of squared gradients (default: 0.9). The `rho` parameter controls the 
+            rate at which the squared gradients are averaged. A higher value of 
+            `rho` will result in a slower average, which can be helpful for preventing 
+            oscillations in the learning process.
```
Comment (janeyx99) [inline torch/optim/adadelta.py:166]:
```suggestion
            of squared gradients (default: 0.9). A higher value of `rho` will
            result in a slower average, which can be helpful for preventing 
            oscillations in the learning process.
```

File: torch/optim/adadelta.py [hunk 8]
```diff
@@ -149,20 +149,36 @@ def step(self, closure=None):
...
+            improve numerical stability.
         lr (float, optional): coefficient that scale delta before it is applied
-            to the parameters (default: 1.0)
-        weight_decay (float, optional): weight decay (L2 penalty) (default: 0)
+            to the parameters (default: 1.0). The `lr` parameter controls the overall
+            learning rate of the optimizer. A higher value of `lr` will result in
+            faster learning, but it can also lead to overfitting.
+        weight_decay (float, optional): weight decay (L2 penalty) (default: 0).
+                      The `weight_decay` parameter applies a penalty to the weights
+                      of the model, which can help to prevent overfitting.
```
Comment (janeyx99) [inline torch/optim/adadelta.py:176]:
```suggestion
```
I don't think overfitting is the problem addressed, but as weight decay is a general concept beyond all optimizers, this commentary doesn't need to be here.

**Review by janeyx99 — COMMENTED** [review]
Sorry for the delay--I forgot to click "comment" the last time I had left these changes. 

Generally, we want to be as concise and precise as possible in our documentation.

**Review by janeyx99 — COMMENTED** [review]
I don't think you meant to remove lines 155 and 164-169. Those should not be removed. Other than that, the PR looks good.

File: torch/optim/adadelta.py [hunk 9]
```diff
@@ -157,16 +157,18 @@ def step(self, closure=None):
...
+            numerical stability (default: 1e-6).
         lr (float, optional): coefficient that scale delta before it is applied
-            to the parameters (default: 1.0)
+                  to the parameters (default: 1.0)
         weight_decay (float, optional): weight decay (L2 penalty) (default: 0)
         {_foreach_doc}
         {_maximize_doc}
         {_differentiable_doc}
-
+
```
Comment (janeyx99) [inline torch/optim/adadelta.py:171]:
```suggestion

```

File: torch/optim/adadelta.py [hunk 10]
```diff
@@ -157,16 +157,18 @@ def step(self, closure=None):
...
-            of squared gradients (default: 0.9)
+            of squared gradients (default: 0.9). A higher value of `rho` will
+            result in a slower average, which can be helpful for preventing 
+            oscillations in the learning process.
         eps (float, optional): term added to the denominator to improve
-            numerical stability (default: 1e-6)
+            numerical stability (default: 1e-6).
         lr (float, optional): coefficient that scale delta before it is applied
-            to the parameters (default: 1.0)
+                  to the parameters (default: 1.0)
```
Comment (janeyx99) [inline torch/optim/adadelta.py:166]:
```suggestion
            to the parameters (default: 1.0)
```

**Comment by hi-sushanta** [comment]
what line? would you please suggest to me?

File: torch/optim/adadelta.py [hunk 11]
```diff
@@ -157,16 +157,18 @@ def step(self, closure=None):
...
-            numerical stability (default: 1e-6)
+            numerical stability (default: 1e-6).
         lr (float, optional): coefficient that scale delta before it is applied
             to the parameters (default: 1.0)
         weight_decay (float, optional): weight decay (L2 penalty) (default: 0)
         {_foreach_doc}
         {_maximize_doc}
         {_differentiable_doc}
-
+
```
Comment (janeyx99) [inline torch/optim/adadelta.py:171]:
```suggestion

```
You've added extra spaces

**Review by hisushanta — COMMENTED** [review]
It's my final version to remove white space.

File: torch/optim/adadelta.py [hunk 12]
```diff
@@ -157,16 +157,17 @@ def step(self, closure=None):
...
-            numerical stability (default: 1e-6)
+            numerical stability (default: 1e-6).
         lr (float, optional): coefficient that scale delta before it is applied
             to the parameters (default: 1.0)
         weight_decay (float, optional): weight decay (L2 penalty) (default: 0)
         {_foreach_doc}
         {_maximize_doc}
-        {_differentiable_doc}
-
+        {_differentiable_doc}
```
Comment (janeyx99) [inline torch/optim/adadelta.py:170]:
```suggestion
        {_differentiable_doc}

```
This should also be unchanged

**Review by janeyx99 — APPROVED** [review]
Yes! Thank you for prevailing haha

File: torch/optim/adadelta.py [hunk 13]
```diff
@@ -157,9 +157,11 @@ def step(self, closure=None):
         params (iterable): iterable of parameters to optimize or dicts defining
             parameter groups
         rho (float, optional): coefficient used for computing a running average
-            of squared gradients (default: 0.9)
+            of squared gradients (default: 0.9). A higher value of `rho` will
+            result in a slower average, which can be helpful for preventing 
+            oscillations in the learning process.
```
Comment (janeyx99) [inline torch/optim/adadelta.py:162]:
```suggestion
            of squared gradients (default: 0.9). A higher value of `rho` will
            result in a slower average, which can be helpful for preventing
            oscillations in the learning process.
```
lint
